exist-core/src/main/java/org/exist/xquery/ModuleContext.java

```diff
@@ -65,14 +65,15 @@ public class ModuleContext extends XQueryContext {
 
     public ModuleContext(final XQueryContext parentContext, final String modulePrefix, final String moduleNamespace,
             final String location) {
-        super();
+        super(parentContext != null ? parentContext.db : null, parentContext != null ? parentContext.getConfiguration() : null, null, false);
```

Comment:
I find this very hard to read. Can this be refactored for readability?

Reply:
@line-o That is not possible in Java as a `super` has to be the first call in the constructor!

Reply:
I did not know that. How about extracting the logic for the first parameter into a method?

Reply:
Maybe reformatting would do the trick?
```java
super(
  parentContext != null ? parentContext.db : null,
  parentContext != null ? parentContext.getConfiguration() : null,
  null,
  false
);
```

Reply:
Digging deeper, I wonder if an additional XQueryContext constructor would make sense.
The module context needs to set its XQueryContext here. This is almost like the constructor that copies the context:
https://github.com/eXist-db/exist/blob/cff6e205ff75f7154d037251a584685e4427b4a3/exist-core/src/main/java/org/exist/xquery/XQueryContext.java#L480 

It does not hand over the profiler and never loads defaults, so it is special.
In which cases does the ModuleContext get instantiated without a parentContext?

Reply:
What happens when we call `super(parentContext)` instead?

Reply:
@line-o I have pushed a commit to reformat the call to the super-constructor as you requested.

> I wonder if an additional XQueryContext constructor would make sense.

I don't see how that would help.

> What happens when we call super(parentContext) instead?

It doesn't work.